From b6d5a57b8181cc2565e2231fc9baf95fc9cc481e Mon Sep 17 00:00:00 2001
From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Sat, 2 Jul 2022 04:18:41 +0900
Subject: [PATCH] core: do not filter out systemd.unit= and run-level specifier
 from kernel command line

Fixes a bug introduced by 846f1da465beda990c1c01346311393f485df467.

The commit 846f1da465beda990c1c01346311393f485df467 made systemd.unit=
filtered out from the command line. That causes debug-generator does not
work as expected on daemon-reexecute, and we cannot call `systemctl
daemon-reexecute` in our test suite running on nspawn.

Fixes issue reported in https://github.com/systemd/systemd/pull/23851#issuecomment-1170992052.

(cherry picked from commit bffde9b5869fffc09e7824d2ac0aeb82a31a134b)

Related: #2087652
---
 src/core/main.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/src/core/main.c b/src/core/main.c
index 667e972364..03efaa03be 100644
--- a/src/core/main.c
+++ b/src/core/main.c
@@ -1812,13 +1812,6 @@ static void filter_args(
                         continue;
                 }
 
-                if (startswith(src[i],
-                               in_initrd() ? "rd.systemd.unit=" : "systemd.unit="))
-                        continue;
-
-                if (runlevel_to_target(src[i]))
-                        continue;
-
                 /* Seems we have a good old option. Let's pass it over to the new instance. */
                 dst[(*dst_index)++] = src[i];
         }
